Source/Particles/PhysicalParticleContainer.H

```diff
@@ -77,6 +77,12 @@ public:
                         const amrex::MultiFab& Bx,
                         const amrex::MultiFab& By,
                         const amrex::MultiFab& Bz) override;
+                        
+    virtual void warpx_copy_attribs(const long* np,
+                            const amrex::Real* xp, const amrex::Real* yp, const amrex::Real* zp,
+                            const amrex::Real* uxp, const amrex::Real* uyp, const amrex::Real* uzp,
+                            amrex::Real* xpold, amrex::Real* ypold, amrex::Real* zpold,
+                            amrex::Real* uxpold, amrex::Real* uypold, amrex::Real* uzpold);
```

Comment:
As such, the function takes a lot of `Real*` arguments. Since `warpx_copy_attribs` is a member function of `PhysicalParticleContainer`, some of them could probably be accessed directly within the function body. That could help readability.

Also, does the function needs to be `virtual`?

Reply:
If you can, I think I would need help to understand this part (again). When I checked this, it seemed to me that these elements were not defined as members of the class. I used virtual because I thought this function was redefined by the derived class RigidInjectedParticleContainer (and that that was the reason to use virtual classes). But of course I can change that!